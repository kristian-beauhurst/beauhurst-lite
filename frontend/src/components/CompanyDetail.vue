<template>
  <div class="company-detail">
    <div v-if="loading" class="loading">
      Loading...
    </div>
    <div v-else-if="error" class="error">
      {{ error }}
    </div>
    <div v-else-if="company" class="content">
      <div class="header">
        <h1>{{ company.name }}</h1>
        <div class="meta">
          <span class="country">{{ company.country?.name }}</span>
          <span class="founded">Founded: {{ formatDate(company.date_founded) }}</span>
          <span class="status" :class="{ active: company.active }">
            {{ company.active ? 'Active' : 'Inactive' }}
          </span>
        </div>
      </div>

      <div class="description">
        <h2>About</h2>
        <p>{{ company.description }}</p>
      </div>

      <div class="employees" v-if="company.employees?.length">
        <h2>Employees ({{ company.employees.length }})</h2>
        <div class="employee-grid">
          <div v-for="employee in company.employees" :key="employee.id" class="employee-card">
            <h3>{{ employee.name }}</h3>
            <p class="job-title">{{ employee.job_title }}</p>
            <p class="contact">
              <a :href="'mailto:' + employee.email">{{ employee.email }}</a>
              <span v-if="employee.phone_number">{{ employee.phone_number }}</span>
            </p>
          </div>
        </div>
      </div>

      <div class="deals" v-if="company.deals?.length">
        <h2>Deals</h2>
        <div class="deals-list">
          <div v-for="deal in company.deals" :key="deal.id" class="deal-card">
            <div class="amount">Â£{{ formatAmount(deal.amount_raised) }}</div>
            <div class="date">{{ formatDate(deal.date_of_deal) }}</div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8000/api/v1';

export default {
  name: 'CompanyDetail',
  props: {
    id: {
      type: [String, Number],
      required: true
    }
  },
  data() {
    return {
      company: null,
      loading: true,
      error: null
    };
  },
  async created() {
    try {
      const response = await axios.get(`${API_BASE_URL}/companies/${this.id}/`);
      this.company = response.data;
    } catch (error) {
      this.error = 'Failed to load company details';
      console.error('Error loading company:', error);
    } finally {
      this.loading = false;
    }
  },
  methods: {
    formatDate(date) {
      if (!date) return 'N/A';
      return new Date(date).toLocaleDateString();
    },
    formatAmount(amount) {
      return new Intl.NumberFormat('en-GB', {
        style: 'decimal',
        minimumFractionDigits: 0,
        maximumFractionDigits: 0
      }).format(amount);
    }
  }
};
</script>

<style scoped>
.company-detail {
  max-width: 1200px;
  margin: 0 auto;
  padding: 2rem;
}

.loading, .error {
  text-align: center;
  padding: 2rem;
  font-size: 1.2rem;
}

.error {
  color: #dc3545;
}

.header {
  margin-bottom: 2rem;
}

.header h1 {
  font-size: 2.5rem;
  color: #2c3e50;
  margin-bottom: 1rem;
}

.meta {
  display: flex;
  gap: 1rem;
  color: #666;
}

.status {
  padding: 0.25rem 0.5rem;
  border-radius: 4px;
  background-color: #dc3545;
  color: white;
}

.status.active {
  background-color: #28a745;
}

.description {
  margin-bottom: 3rem;
}

.description h2 {
  color: #2c3e50;
  margin-bottom: 1rem;
}

.employee-grid {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
  gap: 1.5rem;
  margin-top: 1rem;
}

.employee-card {
  padding: 1.5rem;
  border: 1px solid #ddd;
  border-radius: 8px;
  background-color: white;
}

.employee-card h3 {
  margin: 0 0 0.5rem 0;
  color: #2c3e50;
}

.job-title {
  color: #666;
  margin-bottom: 1rem;
}

.contact {
  font-size: 0.9rem;
}

.contact a {
  color: #007bff;
  text-decoration: none;
}

.contact span {
  display: block;
  color: #666;
}

.deals-list {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
  gap: 1rem;
  margin-top: 1rem;
}

.deal-card {
  padding: 1rem;
  border: 1px solid #ddd;
  border-radius: 8px;
  background-color: white;
}

.amount {
  font-size: 1.2rem;
  font-weight: bold;
  color: #2c3e50;
}

.date {
  color: #666;
  font-size: 0.9rem;
}
</style> 