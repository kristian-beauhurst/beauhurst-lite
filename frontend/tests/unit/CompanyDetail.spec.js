import { mount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'
import CompanyDetail from '@/components/CompanyDetail.vue'
import axios from 'axios'

jest.mock('axios')

const router = createRouter({
  history: createWebHistory(),
  routes: [
    { path: '/companies/:id', component: CompanyDetail }
  ]
})

const mockCompanyData = {
  id: 1,
  name: 'Acme Corporation',
  description: 'A leading technology company specializing in innovative solutions.',
  date_founded: '2010-05-15',
  active: true,
  country: {
    id: 1,
    iso_code: 'GB',
    name: 'United Kingdom'
  },
  employees: [
    {
      id: 1,
      name: 'John Smith',
      job_title: 'Software Engineer',
      email: 'john.smith@acme.com',
      phone_number: '+44 123 456 7890'
    },
    {
      id: 2,
      name: 'Jane Doe',
      job_title: 'Product Manager',
      email: 'jane.doe@acme.com',
      phone_number: '+44 123 456 7891'
    }
  ],
  deals: [
    {
      id: 1,
      amount_raised: 1000000,
      date_of_deal: '2023-01-15'
    },
    {
      id: 2,
      amount_raised: 2500000,
      date_of_deal: '2023-06-20'
    }
  ]
}

describe('CompanyDetail.vue', () => {
  let wrapper

  beforeEach(() => {
    axios.get.mockClear()
  })

  afterEach(() => {
    if (wrapper) {
      wrapper.unmount()
    }
  })

  it('renders loading state correctly', async () => {
    axios.get.mockImplementation(() => new Promise(() => {}))

    wrapper = mount(CompanyDetail, {
      props: { id: 1 },
      global: {
        plugins: [router]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders error state correctly', async () => {
    axios.get.mockRejectedValueOnce(new Error('Network error'))

    wrapper = mount(CompanyDetail, {
      props: { id: 1 },
      global: {
        plugins: [router]
      }
    })

    await wrapper.vm.$nextTick()
    await new Promise(resolve => setTimeout(resolve, 0))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders company data correctly', async () => {
    axios.get.mockResolvedValueOnce({ data: mockCompanyData })

    wrapper = mount(CompanyDetail, {
      props: { id: 1 },
      global: {
        plugins: [router]
      }
    })

    await wrapper.vm.$nextTick()
    await new Promise(resolve => setTimeout(resolve, 0))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders company without employees correctly', async () => {
    const companyWithoutEmployees = {
      ...mockCompanyData,
      employees: []
    }
    axios.get.mockResolvedValueOnce({ data: companyWithoutEmployees })

    wrapper = mount(CompanyDetail, {
      props: { id: 1 },
      global: {
        plugins: [router]
      }
    })

    await wrapper.vm.$nextTick()
    await new Promise(resolve => setTimeout(resolve, 0))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders company without deals correctly', async () => {
    const companyWithoutDeals = {
      ...mockCompanyData,
      deals: []
    }
    axios.get.mockResolvedValueOnce({ data: companyWithoutDeals })

    wrapper = mount(CompanyDetail, {
      props: { id: 1 },
      global: {
        plugins: [router]
      }
    })

    await wrapper.vm.$nextTick()
    await new Promise(resolve => setTimeout(resolve, 0))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders inactive company correctly', async () => {
    const inactiveCompany = {
      ...mockCompanyData,
      active: false
    }
    axios.get.mockResolvedValueOnce({ data: inactiveCompany })

    wrapper = mount(CompanyDetail, {
      props: { id: 1 },
      global: {
        plugins: [router]
      }
    })

    await wrapper.vm.$nextTick()
    await new Promise(resolve => setTimeout(resolve, 0))

    expect(wrapper.html()).toMatchSnapshot()
  })
}) 