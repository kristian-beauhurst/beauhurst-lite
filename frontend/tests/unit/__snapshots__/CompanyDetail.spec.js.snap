// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyDetail.vue renders company data correctly 1`] = `
<div class="company-detail">
  <div class="content">
    <div class="header">
      <h1>Acme Corporation</h1>
      <div class="meta"><span class="country">United Kingdom</span><span class="founded">Founded: 15/05/2010</span><span class="status active">Active</span></div>
    </div>
    <div class="description">
      <h2>About</h2>
      <p>A leading technology company specializing in innovative solutions.</p>
    </div>
    <div class="employees">
      <h2>Employees (2)</h2>
      <div class="employee-grid">
        <div class="employee-card">
          <h3>John Smith</h3>
          <p class="job-title">Software Engineer</p>
          <p class="contact"><a href="mailto:john.smith@acme.com">john.smith@acme.com</a><span>+44 123 456 7890</span></p>
        </div>
        <div class="employee-card">
          <h3>Jane Doe</h3>
          <p class="job-title">Product Manager</p>
          <p class="contact"><a href="mailto:jane.doe@acme.com">jane.doe@acme.com</a><span>+44 123 456 7891</span></p>
        </div>
      </div>
    </div>
    <div class="deals">
      <h2>Deals</h2>
      <div class="deals-list">
        <div class="deal-card">
          <div class="amount">£1,000,000</div>
          <div class="date">15/01/2023</div>
        </div>
        <div class="deal-card">
          <div class="amount">£2,500,000</div>
          <div class="date">20/06/2023</div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CompanyDetail.vue renders company without deals correctly 1`] = `
<div class="company-detail">
  <div class="content">
    <div class="header">
      <h1>Acme Corporation</h1>
      <div class="meta"><span class="country">United Kingdom</span><span class="founded">Founded: 15/05/2010</span><span class="status active">Active</span></div>
    </div>
    <div class="description">
      <h2>About</h2>
      <p>A leading technology company specializing in innovative solutions.</p>
    </div>
    <div class="employees">
      <h2>Employees (2)</h2>
      <div class="employee-grid">
        <div class="employee-card">
          <h3>John Smith</h3>
          <p class="job-title">Software Engineer</p>
          <p class="contact"><a href="mailto:john.smith@acme.com">john.smith@acme.com</a><span>+44 123 456 7890</span></p>
        </div>
        <div class="employee-card">
          <h3>Jane Doe</h3>
          <p class="job-title">Product Manager</p>
          <p class="contact"><a href="mailto:jane.doe@acme.com">jane.doe@acme.com</a><span>+44 123 456 7891</span></p>
        </div>
      </div>
    </div>
    <!--v-if-->
  </div>
</div>
`;

exports[`CompanyDetail.vue renders company without employees correctly 1`] = `
<div class="company-detail">
  <div class="content">
    <div class="header">
      <h1>Acme Corporation</h1>
      <div class="meta"><span class="country">United Kingdom</span><span class="founded">Founded: 15/05/2010</span><span class="status active">Active</span></div>
    </div>
    <div class="description">
      <h2>About</h2>
      <p>A leading technology company specializing in innovative solutions.</p>
    </div>
    <!--v-if-->
    <div class="deals">
      <h2>Deals</h2>
      <div class="deals-list">
        <div class="deal-card">
          <div class="amount">£1,000,000</div>
          <div class="date">15/01/2023</div>
        </div>
        <div class="deal-card">
          <div class="amount">£2,500,000</div>
          <div class="date">20/06/2023</div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CompanyDetail.vue renders error state correctly 1`] = `
<div class="company-detail">
  <div class="error">Failed to load company details</div>
</div>
`;

exports[`CompanyDetail.vue renders inactive company correctly 1`] = `
<div class="company-detail">
  <div class="content">
    <div class="header">
      <h1>Acme Corporation</h1>
      <div class="meta"><span class="country">United Kingdom</span><span class="founded">Founded: 15/05/2010</span><span class="status">Inactive</span></div>
    </div>
    <div class="description">
      <h2>About</h2>
      <p>A leading technology company specializing in innovative solutions.</p>
    </div>
    <div class="employees">
      <h2>Employees (2)</h2>
      <div class="employee-grid">
        <div class="employee-card">
          <h3>John Smith</h3>
          <p class="job-title">Software Engineer</p>
          <p class="contact"><a href="mailto:john.smith@acme.com">john.smith@acme.com</a><span>+44 123 456 7890</span></p>
        </div>
        <div class="employee-card">
          <h3>Jane Doe</h3>
          <p class="job-title">Product Manager</p>
          <p class="contact"><a href="mailto:jane.doe@acme.com">jane.doe@acme.com</a><span>+44 123 456 7891</span></p>
        </div>
      </div>
    </div>
    <div class="deals">
      <h2>Deals</h2>
      <div class="deals-list">
        <div class="deal-card">
          <div class="amount">£1,000,000</div>
          <div class="date">15/01/2023</div>
        </div>
        <div class="deal-card">
          <div class="amount">£2,500,000</div>
          <div class="date">20/06/2023</div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CompanyDetail.vue renders loading state correctly 1`] = `
<div class="company-detail">
  <div class="loading"> Loading... </div>
</div>
`;
